polkadot-protocol/protocol-components/accounts/account-data-structure.md

```diff
@@ -0,0 +1,132 @@
+
+!!! note
+    See [StorageMap API](https://paritytech.github.io/polkadot-sdk/master/frame_support/storage/types/struct.StorageMap.html) for more details.
+
+## Account Info
+
+The `AccountInfo` for an account is also defined within the [`frame-system`](https://paritytech.github.io/polkadot-sdk/master/src/frame_system/lib.rs.html) pallet:
+
+```rs
+#[derive(Clone, Eq, PartialEq, Default, RuntimeDebug, Encode, Decode)]
+pub struct AccountInfo<Nonce, AccountData> {
+  /// The number of transactions this account has sent.
+  pub nonce: Nonce,
+  /// The number of other modules that currently depend on this account's existence. The account
+  /// cannot be reaped until this is zero.
+  pub consumers: RefCount,
+  /// The number of other modules that allow this account to exist. The account may not be reaped
+  /// until this and `sufficients` are both zero.
+  pub providers: RefCount,
+  /// The number of modules that allow this account to exist for their own purposes only. The
+  /// account may not be reaped until this and `providers` are both zero.
+  pub sufficients: RefCount,
+  /// The additional data that belongs to this account. Used to store the balance(s) in a lot of
+  /// chains.
+  pub data: AccountData,
+}
+```
+
+Each account has an `AccountInfo` structure that includes the following components:
+
+- `nonce`: Tracks the number of transactions the account has sent.
+- `consumers`: A reference counter indicating the number of other modules that depend on this account's existence.
+- `providers`: A reference counter indicating the number of modules that permit this account to exist.
+- `sufficients`: A reference counter indicating the number of modules that allow this account to exist for their own purposes only.
+- `AccountData`: A configurable structure that can store various types of data specific to the account.
+
+## Account Reference Counters
+
+Account reference counters track dependencies in the runtime. For instance, if data is stored in a map associated with an account, you wouldn't want to delete the account until the data under its control has been removed.
+
+The `consumers` and `providers` reference counters are intended to work together. For example, in the [`session`](https://docs.rs/pallet-session/latest/pallet_session/) pallet, the consumers counter is incremented when an account sets its session keys before becoming a validator. 
+
+The providers counter must be greater than zero for the consumers counter to be incremented.
+
+### Providers Reference Counters
+
+The `providers` reference counter indicates whether an account is ready to be depended upon. For example, it is incremented when a new account is created with a balance greater than the existential deposit.
+
+The `providers` reference counter prevents Substrate pallets from storing data about an account until the account is active (`providers` > 0)
+
+### Consumers Reference Counters
```

Comment:
Not sure if necessary here, but maybe we have a bullet list explaining what a provider, consumer, and sufficient is in this context?

Reply:
Not sure I understand what to add here, could you give me more context?

Reply:
Ah nvm, seems to be a list above that I missed, all good now!